lib/EntryDownloader.js

```diff
@@ -85,28 +86,41 @@ var EntryDownloader = (function () {
             });
         });
         // Get the destination to save the manifest
-        var destPath = path.join(config.get('rootFolderPath'), entryId, persistenceFormat.getMasterManifestName());
+        var destPath = path.join(config.get('rootFolderPath'), that.entryId, persistenceFormat.getMasterManifestName());
         var playerMasterManifest = playlist.toString();
-        logger.info("Entry: " + entryId + "received master manifest:\n" + playerMasterManifest);
+        logger.info("Entry: %s received master manifest: %j", that.entryId, that.flavorsObjArray);
         return fsUtils.writeFileAtomically(destPath, playerMasterManifest)
             .then(function() {
-                return flavorsObjArray;
+                that.lastPlaylistUpdate = new Date();
             });
     };
 
+    EntryDownloader.prototype.updatePlaylist = function() {
+        var currTime = new Date();
+        // Update the entry's playlist every minute
+        if ((currTime - this.lastPlaylistUpdate) > config.get('playlistUpdateInterval')) {
```

Comment:
should be  (!this.lastPlaylistUpdate || ((currTime - this.lastPlaylistUpdate) > config.get('playlistUpdateInterval'))